client-src/elements/chromedash-form-field.js

```diff
@@ -91,15 +106,49 @@ export class ChromedashFormField extends LitElement {
 
     // Dispatch a new event to notify other components of the changes.
     this.dispatchEvent(new CustomEvent('form-field-update', eventOptions));
+
+    this.doSemanticChecks(this.fieldProps.check);
+  }
+
+  doSemanticChecks(checkFunction) {
+    if (checkFunction) {
+      const fieldValue = this.getValue();
+      const checkResult = checkFunction(fieldValue,
+        (name) => getFieldValue(name, this.fieldValues));
+      if (checkResult != null) {
+        this.checkMessage = html`
+          <span class="check-${
+              checkResult.message ? 'message' :
+              checkResult.warning ? 'warning' :
+              checkResult.error ? 'error' : 'unknown'
+            }">
+            ${
+              checkResult.message ? checkResult.message :
+              checkResult.warning ? html`<b>Warning</b>: ${checkResult.warning}` :
+              checkResult.error ? html`<b>Error</b>: ${checkResult.error}` :
+              ''
+            }
+          </span>`;
+
+        const formFieldElement = this.renderRoot.querySelector(`#id_${this.name}`);
+        if (formFieldElement && formFieldElement.setCustomValidity &&
```

Comment:
Th first two parts of this condition could be combined to a little shorter as `formFieldElement?.setCustomValidity`

Reply:
Right.  Good point.